include/dlr_common.h

```diff
@@ -0,0 +1,100 @@
+#ifndef DLR_COMMON_H_
+#define DLR_COMMON_H_
+
+#include <string>
+#include <vector>
+#include <sys/types.h>
+#include <dmlc/logging.h>
+#include <runtime_base.h>
+
+#define LINE_SIZE 256
+
+#ifndef _WIN32
+
+/* For basename() on UNIX */
+#include <libgen.h>
+
+#endif
+
+/* OS-specific library file extension */
+#ifdef _WIN32
+#define LIBEXT ".dll"
+#elif __APPLE__
+#define LIBEXT ".dylib"
+#else
+#define LIBEXT ".so"
+#endif
+
+namespace dlr {
+
+/* The following file names are reserved by SageMaker and should not be used
+ * as model JSON */
+constexpr const char* SAGEMAKER_AUXILIARY_JSON_FILES[] = {
+  "model-shapes.json", "hyperparams.json"
+};
+
+typedef struct {
+  std::string model_lib;
+  std::string params;
+  std::string model_json;
+  std::string ver_json;
+} ModelPath;
+
+void listdir(const std::string& dirname, std::vector<std::string> &paths);
+
+std::string GetBasename(const std::string& path);
+
+inline bool endsWith(const std::string &mainStr, const std::string &toMatch) {
+	if (mainStr.size() >= toMatch.size() &&
+	  mainStr.compare(mainStr.size() - toMatch.size(), toMatch.size(), toMatch) == 0)
+		return true;
+	else
+		return false;
+}
+
+enum class DLRBackend {
+  kTVM,
+  kTREELITE
+};
+
+/*! /brief Get the backend based on the contents of the model folder.
+ */
+DLRBackend get_backend(const std::string &dirname);
```

Comment:
Let's use CamelCase for function names to keep consistency.

`DLRBackend GetBackend(const std::string& dirname);`

Reply:
fixed